src/packet.c

```
@@ -3,295 +3,307 @@
+    replay_tracker_t* tr = &packet_state.replay_trackers[idx];
+
+    /* Sequence ahead of highest seen: accept and advance window */
+    if (seq > tr->highest_seen_seq) {
+        uint32_t delta = seq - tr->highest_seen_seq;
+        if (delta >= REPLAY_WINDOW_SIZE) {
+            /* Big jump — reset window */
+            tr->window_bits = 0ULL;
+        } else {
+            /* Shift window left, fill bits for newly accepted range.
+             * Mask to avoid shifting beyond 64 bits when delta is small but close to limit. */
+            uint64_t all_ones = (REPLAY_WINDOW_SIZE >= 64) ? ~0ULL : ((1ULL << REPLAY_WINDOW_SIZE) - 1);
```

Cppcheck (2.21.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)